services/s3/src/test/java/software/amazon/awssdk/services/s3/S3EndpointResolutionTest.java

```diff
@@ -508,4 +512,100 @@ private S3Configuration withDualstackAndAccelerateEnabled() {
+    }
+
+    @Test
+    public void accessPointArn_us_gov_west_1_clientRegion_fips_us_gov_east_1_useArnRegionTrue() throws Exception {
+        mockHttpClient.stubNextResponse(mockListObjectsResponse());
+        S3Client s3Client = clientBuilder().region(Region.of("fips-us-gov-east-1"))
+                .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build()).build();
+        String accessPointArn = "arn:aws-us-gov:s3:us-gov-west-1:123456789012:accesspoint:myendpoint";
+
+        assertThatThrownBy(() -> s3Client.getObject(GetObjectRequest.builder().bucket(accessPointArn).key("someKey").build()))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(CROSS_REGION_FIPS_ERROR);
+    }
+
+    @Test
+    public void accessPointArn_us_gov_west_1_clientRegion_fips_us_gov_east_1_useArnRegionFalse() throws Exception {
+        mockHttpClient.stubNextResponse(mockListObjectsResponse());
+        S3Client s3Client = clientBuilder().region(Region.of("fips-us-gov-east-1"))
+                .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(false).build()).build();
+        String accessPointArn = "arn:aws-us-gov:s3:us-gov-west-1:123456789012:accesspoint:myendpoint";
+
+        assertThatThrownBy(() -> s3Client.getObject(GetObjectRequest.builder().bucket(accessPointArn).key("someKey").build()))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(CROSS_REGION_FIPS_ERROR);
+    }
+
+    @Test
+    public void accessPointArnRegion_fips_clientRegion_non_fips_useArnRegionTrue() throws Exception {
+        mockHttpClient.stubNextResponse(mockListObjectsResponse());
+        S3Client s3Client = clientBuilder().region(Region.of("us-west-1"))
+                .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build()).build();
+        String accessPointArn = "arn:aws-us-gov:s3:fips-us-gov-east-1:123456789012:accesspoint:myendpoint";
+
+        assertThatThrownBy(() -> s3Client.getObject(GetObjectRequest.builder().bucket(accessPointArn).key("someKey").build()))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining(CROSS_REGION_FIPS_ERROR);
+    }
+
+
+    @Test
+    public void accessPointArn_non_region_fips_useArnRegionTrue() throws Exception {
+        mockHttpClient.stubNextResponse(mockListObjectsResponse());
+        S3Client s3Client = clientBuilder().region(Region.of("us-west-2-fips"))
+                .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build()).build();
+        String accessPointArn = "arn:aws:s3:us-west-2:12345678910:accesspoint:foobar";
+        s3Client.getObject(GetObjectRequest.builder().bucket(accessPointArn).key("someKey").build());
+        assertThat(mockHttpClient.getLastRequest().getUri().getHost())
+                .isEqualTo("foobar-12345678910.s3-accesspoint.us-west-2.amazonaws.com");
+
+    }
```

Comment:
1. Shouldn't this resolve to the FIPS endpoint? The customer has configured the client for FIPS, I thought the point of these changes were to make sure that if the customer thinks they're using FIPS that we shouldn't ever use a non-FIPS endpoint because it might result in the customer accidentally being uncompliant with FIPS.
2. This wasn't really the test I was asking for. I was interested in seeing where the ARN region is us-west-2-fips, not the client region. This is no different than scenario 5.

Reply:
Thanks Matt , I was too focused on Not allowing the request if mismatch and missed to observe the resolved endpoints for this case.


1. Yeap I looked at the original table and found that it should actually resolve to fips
The issue is because of [this code](https://github.com/aws/aws-sdk-java-v2/blob/584ccb59e770177aeaa4c3b6bda4e24015b8ece9/services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/S3EndpointUtils.java#L56-L61) where it says isFipsRegionProvided is false if useArnRegion  is true and  Arn not in fips and even though client is in fips
Will correct this.

2.  The actual spec table does not mention about the case where to resolve the endpoints if Client is Non fips and Arn is in fips. I will try to get the requirement.
3. 



```
    // "arn:aws:s3:fips-us-west-1:123456789012:accesspoint:myendpoint"
    //"us-west-1"
    // "myendpoint-123456789012.s3-accesspoint.us-west-1.amazonaws.com"
    // "myendpoint-123456789012.s3-accesspoint.fips.us-west-1.amazonaws.com"   
```

Reply:
Thanks Matt , updated the PR with the fips endpoint.